src/script/interpreter.cpp

```diff
@@ -467,6 +467,47 @@ static bool EvalChecksigPreTapscript(const valtype& vchSig, const valtype& vchPu
     return true;
 }
 
+static bool EvalTapScriptCheckSigFromStack(const valtype& sig, const XOnlyPubKey& pubkey, ScriptExecutionData& execdata, unsigned int flags, const Span<const unsigned char> msg, SigVersion sigversion, ScriptError* serror, bool& success)
+{
+    // This code follows the behaviour of EvalCheckSigTapscript
+    assert(sigversion == SigVersion::TAPSCRIPT);
+
+    /*
+     *  The following validation sequence is consensus critical. Please note how --
+     *    upgradable public key versions precede other rules;
+     *    the script execution fails when using empty signature with invalid public key;
+     *    the script execution fails when using non-empty invalid signature.
+     */
+    success = !sig.empty();
+    if (success) {
+        // Implement the sigops/witnesssize ratio test.
+        // Passing with an upgradable public key version is also counted.
+        assert(execdata.m_validation_weight_left_init);
+        execdata.m_validation_weight_left -= VALIDATION_WEIGHT_PER_SIGOP_PASSED;
+        if (execdata.m_validation_weight_left < 0) {
+            return set_error(serror, SCRIPT_ERR_TAPSCRIPT_VALIDATION_WEIGHT);
+        }
+    }
+    if (pubkey.size() == 0) {
+        return set_error(serror, SCRIPT_ERR_PUBKEYTYPE);
+    } else if (pubkey.size() == 32) {
+        if (success && !pubkey.VerifySchnorr(msg, sig)) {
```

Comment:
`VerifySchnorr` has a precondition that `sig` is 64-bytes, but I don't see a check for this precondition.

More testing?

Reply:
Indeed, I tested empty sigs, mutated 64-byte sigs, but did not check 64-byte sigs :(

Reply:
:/ surprised the fuzzer didn't hit this given that it's an actual assertion failure and not even guarded by any cryptography